test/integ/routers/alpha-router/alpha-router.integration.test.ts

```diff
@@ -0,0 +1,1133 @@
+
+          const {
+            quote,
+            quoteDecimals,
+            quoteGasAdjustedDecimals,
+            methodParameters,
+          } = parseSwap(swap!);
+
+          await validateStandardSwapRoute(
+            quoteDecimals,
+            quoteGasAdjustedDecimals,
+            tradeType,
+            100,
+            10
+          );
+
+          await validateExecuteSwap(
+            quote,
+            tokenIn,
+            tokenOut,
+            methodParameters,
+            tradeType,
+            '100',
+            '100'
+          );
+        });
+
+        it(`erc20 -> eth`, async () => {
+          const tokenIn = USDC_MAINNET;
+          const tokenOut = Ether.onChain(1) as Currency;
+          const amount =
+            tradeType == TradeType.EXACT_INPUT
+              ? parseAmount('1000000', tokenIn)
+              : parseAmount('10', tokenOut);
+
+          const swap = await alphaRouter.route(
+            amount,
+            getQuoteToken(tokenIn, tokenOut, tradeType),
+            tradeType,
+            {
+              recipient: alice._address,
+              slippageTolerance: SLIPPAGE,
+              deadline: parseDeadline(360),
+            },
+            {
+              ...ROUTING_CONFIG,
+            }
+          );
+          expect(swap).toBeDefined();
+          expect(swap).not.toBeNull();
+
+          const {
+            quote,
+            quoteDecimals,
+            quoteGasAdjustedDecimals,
+            methodParameters,
+          } = parseSwap(swap!);
+
+          await validateStandardSwapRoute(
+            quoteDecimals,
+            quoteGasAdjustedDecimals,
+            tradeType
+          );
+
+          await validateExecuteSwap(
+            quote,
+            tokenIn,
+            tokenOut,
+            methodParameters,
+            tradeType,
+            '1000000'
+          );
+        });
+
+        it(`erc20 -> eth large trade`, async () => {
+          // Trade of this size almost always results in splits.
+          const tokenIn = USDC_MAINNET;
+          const tokenOut = Ether.onChain(1) as Currency;
+          const amount =
+            tradeType == TradeType.EXACT_INPUT
+              ? parseAmount('1000000', tokenIn)
+              : parseAmount('100', tokenOut);
+
+          const swap = await alphaRouter.route(
+            amount,
+            getQuoteToken(tokenIn, tokenOut, tradeType),
+            tradeType,
+            {
+              recipient: alice._address,
+              slippageTolerance: SLIPPAGE,
+              deadline: parseDeadline(360),
+            },
+            {
+              ...ROUTING_CONFIG,
+            }
+          );
+          expect(swap).toBeDefined();
+          expect(swap).not.toBeNull();
+
+          const { quote, methodParameters } = parseSwap(swap!);
+
+          const { route } = getRouteFromSwap(tradeType, swap!);
+
+          expect(route).not.toBeUndefined;
+
+          const amountInEdgesTotal = _(route)
```

Comment:
small, but you could also just calculate the edgeAmounts as you're iterating through the pools in `getRouteFromSwap`

Reply:
true, though in this case we're building up the `amountInEdgesTotal` to check that it's equal to `amountIn`, and I think it would be better to keep the `getRouteFromSwap` as just a helper function that transforms the data